### ISSUE TITLE
About Single Image Super-Resolution

### ISSUE THREAD

**Issue by farmingyard** [dzld00/pytorch-manifold-matching#2]
Great job!
Could you please share some sample codes about Single Image Super-Resolution setting (A)? Thank you very much.

**Comment by dzld00** [comment]
Thanks for your interest in the work! We haven't got a change to organize the code for SR experiments yet. For fast implementation, one way is to use the template at https://github.com/leftthomas/SRGAN and simply replace the loss functions and parameter settings in the paper.

**Comment by farmingyard** [comment]
Thanks for your reply.
I have the standard SR codes yet，but what I need is how to replace the gan loss with your method.
Now I train a 1X SR network with G loss and D loss alternately, the G loss is L1 pixel loss(loss weight=1.0) and your MM loss(loss weight=1e-3, No use your pair loss yet for simplify), the D loss is your Triplet loss(loss weight=1.0, margin=1.0, alpha=1e-2),  the Triplet embedding network is just as your paper supplement, when I check the results with standard gan network,  they are not sharp enough,  so I wondered if my setting is right? How to get more sharpen results just like gan network?
Waiting for your reply,  thank you.

**Comment by dzld00** [comment]
Hello不知道我理解的对不对，问题在1或2 里面:
1. 用Triplet network作为GAN的Discriminator出来的效果不sharp: Triplet network在vanilla GAN的framework里应该不能拿来直接作为Discriminator用;
2. 把GAN loss和Discriminator network换成MvM loss和Triplet network, 效果不sharp: 在SR这个task里面我们观察到的GAN和MvM loss的behavior不太一样, 图9确实也不咋sharp, 我们没有做过只用MvM loss的情况下关于sharpness的对比实验, 这也是一个open question。想要sharp的话可以尝试match higher frequency features, 增加 loss weights, 借助GAN和VGG perceptual loss等等方式。

有问题欢迎随时讨论 dzld00511@126.com
